src/main/java/com/synopsys/integration/blackduck/codelocation/signaturescanner/command/ToolsApiScannerInstaller.java

```diff
@@ -194,6 +201,10 @@ protected String downloadSignatureScanner(File scannerExpansionDirectory, HttpUr
                 oneJar.setExecutable(true);
                 scanExecutable.setExecutable(true);
 
+                if (connectAndGetServerCertificate(downloadUrl)!= null) {
```

Comment:
It would be good to not call this redundantly: store the return value in a variable and use it for both the if condition and the update call.

Reply:
I am not clear on how to do that since the download call (to _update_ the scan-cli) does not explicitly establish a connection here for me to hold on to (I'm not entirely clear on its lifetime either, though I believe it is until the end of the try-with-resources block).

Reply:
What I had in mind is:


```
Certificate cert = connectAndGetServerCertificate(downloadUrl);

if (cert != null) {
    keyStoreHelper.updateKeyStoreWithServerCertificate(downloadUrl.url().getHost(), cert, scanPaths.getPathToCacerts());
}
```

Reply:
Oh I misunderstood your original comment. Thanks for noticing this, done!